### ISSUE TITLE
Naming of TF conditionals

### ISSUE THREAD

**Issue by muzimuzhi** [latex3/latex3#1634]
In reviewing #1622 I suggested the renaming from `\regex_match:(n|N)TF` to `\regex_if_match:...`, see https://github.com/latex3/latex3/pull/1622#discussion_r1860958658.

Recently I found more cases in `\peek_` functions, like `\peek_meaning:NTF`, so I did an overall research on naming schemes of TF conditionals.

| Naming scheme | Example(s) | Note |
|--------|--------|--------|
| `\<type>_if...:...TF` | `\tl_if_empty:NTF`, `\legacy_if:NTF` | Most TF conditionals belong to this type. |
| `\<type>_compare...:...TF` | `\int_compare:nNnTF` |  |
| `\<type>_case:...TF` | `\str_case:nnTF` |  |
| `\bool_...:...TF` | `\bool_lazy_and:nnTF` |  |
| `\<type>_<verb>...:...TF` | `\clist_get:NNTF` | The no-TF versions always exist. [1] |
| Other | `\regex_match:NnTF` | The no-TF versions don't exist. [2] | 

Long notes
- [1] The false code of `\<type>_<verb>...:...TF` is used when the action denoted by `<verb>` would fail, e.g., popping from an empty compound-data variable.
- [2] For the last type, there are
  ```tex
  \peek_catcode:NTF
  \peek_catcode_remove:NTF
  \peek_charcode:NTF
  \peek_charcode_remove:NTF
  \peek_meaning:NTF
  \peek_meaning_remove:NTF
  \peek_regex:NTF
  \peek_regex:nTF
  \peek_regex_remove_once:NTF
  \peek_regex_remove_once:nTF
  \regex_match:NnTF
  \regex_match:nnTF
  ```
  Maybe they all could be renamed to the `\<type>_if...:...TF` form, i.e., following the first scheme.

**Comment by muzimuzhi** [comment]
I used `expl3.cwl` as the data of all l3kernel variables and functions (including their variants). It's generated by the [`latex3command.py` script](https://github.com/James-Yu/LaTeX-Workshop/blob/3b0ae3ae80fdd20377f4fe8fe23094d4b620086f/dev/latex3command.py) from  repository of LaTeX-Workshop VScode extension. LaTeX-Workshop uses `expl3.cwl` as the intermediate data for the final completion list [`expl3.json`](https://github.com/James-Yu/LaTeX-Workshop/blob/3b0ae3ae80fdd20377f4fe8fe23094d4b620086f/data/packages/expl3.json).

Steps to generate `expl3.cwl` yourself:
```
$ git clone --depth=1 git@github.com:James-Yu/LaTeX-Workshop.git
$ cd LaTeX-Workshop/dev
$ export PATH_TO_DTX=path/to/l3kernel/dtx
# requires pip package "requests"
$ python3 latex3command.py
```

`expl3.cwl` generated from l3kernel dtx, at release 2024-11-02.
[expl3.cwl.txt](https://github.com/user-attachments/files/18053803/expl3.cwl.txt)

**Comment by josephwright** [comment]
The `peek` ones don't really make sense if you add `if` since `peek` is itself a verb - `\peek_if_catcode:NTF` is to me confusing as the peeking always happens, but the `if` suggests it's not always applied.

The `regex` case is different - `regex_if_match` isn't confusing.

**Comment by FrankMittelbach** [comment]
I agree with @josephwright and I would also not change `_compare` , `_case`, or `_get`.
What remains for me are the boolean ones, e.g. `\bool_if_lazy_and...` and the like.

**Comment by muzimuzhi** [comment]
> The `peek` ones don't really make sense if you add `if` since `peek` is itself a verb - `\peek_if_catcode:NTF` is to me confusing as the peeking always happens, but the `if` suggests it's not always applied.

@josephwright Nice argument. I'm persuaded. I recalled I once noticed that "`peek` is itself a verb" too thus is special, but forgot it when typing the issue description.

**Comment by car222222** [comment]
Note that "peek" is also a noun, as in 
"Take a peek at . . ."

But this is probably not relevant!
